govcd/adminorg_ldap_test.go

```diff
@@ -0,0 +1,312 @@
+// +build user functional ALL
+
+/*
+ * Copyright 2020 VMware, Inc.  All rights reserved.  Licensed under the Apache v2 License.
+ */
+
+package govcd
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+	. "gopkg.in/check.v1"
+)
+
+// Test_LDAP serves as a "subtest" framework for tests requiring LDAP configuration. It sets up LDAP
+// server and configuration for Org and cleans up this test run.
+//
+// Prerequisites:
+// * External network subnet must have access to internet
+// * Correct DNS servers must be set for external network so that guest VM can resolve DNS records
+func (vcd *TestVCD) Test_LDAP(check *C) {
+	fmt.Println("Setting up LDAP")
+	networkName, vappName, vmName := vcd.configureLdap(check)
+	defer func() {
+		fmt.Println("Unconfiguring LDAP")
+		vcd.unconfigureLdap(check, networkName, vappName, vmName)
+	}()
+
+	// Run tests requiring LDAP from here.
+	vcd.test_GroupCRUD(check)
+	vcd.test_GroupFinderGetGenericEntity(check)
+
+}
+
+// configureLdap creates direct network, spawns Photon OS VM with LDAP server and configures vCD to
+// use LDAP server
+func (vcd *TestVCD) configureLdap(check *C) (string, string, string) {
+	if vcd.skipAdminTests {
+		check.Skip(fmt.Sprintf(TestRequiresSysAdminPrivileges, check.TestName()))
```

Comment:
Could we do check to if external network is configured right and skip test otherwise. This would save testing time(spawning photon with LDAP).

Reply:
This is not really possible because it may have any configuration and I have no obvious way to cover all cases. It could even not have DNS set, but the default ones in guest could work. Besides it would mask the test run then and one could think - "I have run all the tests" while in fact some where skipped.